pkg/processor/pod/pod.go

```diff
@@ -73,14 +74,23 @@ func ProcessSpec(objName string, appMeta helmify.AppMetadata, spec corev1.PodSpe
 	}
 
 	// process nodeSelector if presented:
-	if len(spec.NodeSelector) != 0 {
+	if spec.NodeSelector == nil {
+		return specMap, values, nil
+	} else {
 		err = unstructured.SetNestedField(specMap, fmt.Sprintf(`{{- toYaml .Values.%s.nodeSelector | nindent 8 }}`, objName), "nodeSelector")
 		if err != nil {
 			return nil, nil, err
 		}
-		err = unstructured.SetNestedStringMap(values, spec.NodeSelector, objName, "nodeSelector")
-		if err != nil {
-			return nil, nil, err
+		if len(spec.NodeSelector) != 0 {
```

Comment:
map len check is effectively equal to nil check - an uninitialized map will return zero len.

Reply:
removed it.